port/unix/omrsysinfo.c

```diff
@@ -1967,7 +1974,165 @@ omrsysinfo_get_linux_aarch64_description(struct OMRPortLibrary *portLibrary, OMR
 
 	return 0;
 }
-#endif /* defined(AARCH64) && defined(LINUX) */
+#elif defined(OSX)
+/**
+ * @internal
+ * Populates OMRProcessorDesc *desc on macOS ARM64
+ *
+ * @param[in] desc pointer to the struct that will contain the CPU type and features.
+ *
+ * @return 0 on success, -1 on failure
+ */
+static intptr_t
+omrsysinfo_get_macos_aarch64_description(struct OMRPortLibrary *portLibrary, OMRProcessorDesc *desc)
+{
+	int64_t val;
+	size_t size = sizeof(val);
+
+	desc->processor = OMR_PROCESSOR_ARM64_V8_A;
+	desc->physicalProcessor = desc->processor;
+	desc->features[0] = 0;
+	desc->features[1] = 0;
+
+	/*
+	 * These features are standard on Apple Silicon
+	 * No need to check
+	 */
+	omrsysinfo_set_feature(desc, OMR_FEATURE_ARM64_FP);
+	omrsysinfo_set_feature(desc, OMR_FEATURE_ARM64_ASIMD);
+
+	/*
+	 * Advanced SIMD and Floating Point Capabilities
+	 */
+	if (sysctlbyname("hw.optional.arm.FEAT_BF16", &val, &size, NULL, 0) == 0 &&
+	    val != 0) {
+		omrsysinfo_set_feature(desc, OMR_FEATURE_ARM64_BF16);
+	}
+
+	if (sysctlbyname("hw.optional.arm.FEAT_DotProd", &val, &size, NULL, 0) == 0 &&
+	    val != 0) {
+		omrsysinfo_set_feature(desc, OMR_FEATURE_ARM64_DOTPROD);
+	}
+
+	if (((sysctlbyname("hw.optional.arm.FEAT_FCMA", &val, &size, NULL, 0) == 0) ||
+	     (sysctlbyname("hw.optional.armv8_3_compnum", &val, &size, NULL, 0) == 0)) &&
```

Comment:
Some attribute names for `sysctlbyname` are different depending on macOS versions.
`hw.optional.arm.FEAT_FCMA` is available from macOS 12, while `hw.optional.armv8_3_compnum` is available only on macOS 11.

Reply:
Added comments "Attribute name changed in macOS 12."